src/test/java/org/folio/rest/api/InstanceBulkTest.java

```diff
@@ -0,0 +1,400 @@
+            if (r.succeeded()) {
+              sequenceSet.complete(null);
+            } else {
+              sequenceSet.completeExceptionally(r.cause());
+            }
+          });
+    });
+
+    try {
+      sequenceSet.get(2, SECONDS);
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private void createInstance(JsonObject instanceToCreate)
+      throws MalformedURLException,
+      InterruptedException,
+      ExecutionException,
+      TimeoutException {
+
+    CompletableFuture<Response> createCompleted = new CompletableFuture<>();
+
+    client.post(instancesStorageUrl(""), instanceToCreate,
+      TENANT_ID, json(createCompleted));
+
+    Response response = createCompleted.get(2, SECONDS);
+
+    assertThat(String.format("Create instance failed: %s", response.getBody()),
+      response.getStatusCode(), is(201));
+  }
+
+  private void createManyMoons(Map<String, JsonObject> instancesToCreate)
+      throws MalformedURLException,
+      InterruptedException,
+      ExecutionException,
+      TimeoutException {
+
+    for (String id : instancesToCreate.keySet()) {
+      createInstance(instancesToCreate.get(id));
+    }
+  }
+
+  private Map<String, JsonObject> manyMoons(int total) {
+    return manyMoons(total, InstanceBulkIdsGetField.ID);
+  }
+
+  private Map<String, JsonObject> manyMoons(int total,
+      InstanceBulkIdsGetField field) {
+    Map<String, JsonObject> moons = new HashMap<String, JsonObject>();
+
+    List<String[]> languages = Arrays.asList(new String[][] {
+      new String[] {"eng"},
+      new String[] {"ger"},
+      new String[] {"por", "ger"}
+    });
+
+    for (int i = 0; i < total; i++) {
+      UUID uuid = UUID.randomUUID();
+      String hrid = String.format("in%011d", i + 1);
+
+      String id = "";
+      if (field.compareTo(InstanceBulkIdsGetField.ID) == 0) {
+        id = uuid.toString();
+      }
+
+      JsonArray identifiers = new JsonArray();
+      identifiers.add(identifier(UUID_ISBN, "9781473619777"));
+      JsonArray contributors = new JsonArray();
+      contributors.add(contributor(UUID_PERSONAL_NAME, "Chambers, Becky"));
+      JsonArray tags = new JsonArray();
+      tags.add("test-tag");
+
+      JsonArray langs = new JsonArray();
+      int l = 0;
+      if (i % 3 == 0) {
+        l = 2;
+      }
+      else if (i % 2 == 0) {
+        l = 1;
+      }
+      for (int j = 0; j < languages.get(l).length; j++) {
+        langs.add(languages.get(l)[j]);
+      }
+
+      JsonObject jsonb = createInstanceRequest(uuid, "TEST", "Moon #" + i,
+        identifiers, contributors, UUID_INSTANCE_TYPE, tags, hrid, langs);
+      moons.put(id, jsonb);
+    }
+
+    return moons;
+  }
+
+  public Buffer bufferedCheckURLs(TestContext context, URL url,
+      int codeExpected) {
+    String accept = "application/json";
+    return bufferedCheckURLs(context, url, codeExpected, accept);
+  }
+
+  private Buffer bufferedCheckURLs(TestContext context, URL url,
```

Comment:
Can it be replaced with the default `OkapiClient` and `ResponseHandler.any`:
```
CompletableFuture<Response> getCompleted = new CompletableFuture<>();
client.get(url, TENANT_ID, ResponseHandler.any(getCompleted));

Response response = getCompleted.get(5, SECONDS);

assertThat(response.getStatusCode(), is(codeExpected));
assertThat(response.getContentType(), is(accept));
```

Reply:
The reason for these tests were to do an explicit stream test.
If the explicit buffered stream test is not needed, then the tests like `canGetInstanceBulkOfIdStream` could be removed entirely (being that they are otherwise redundant).

Is there a way to explicitly test that the stream is in fact streaming or is that unnecessary at this scope?
Should I remove this explicit stream entirely?

Reply:
Testing whether PgUtil.streamGet is implemented correctly is unnecessary at this scope. Such tests should go into RMB's [PgUtilIT](https://github.com/folio-org/raml-module-builder/blob/master/domain-models-runtime/src/test/java/org/folio/rest/persist/PgUtilIT.java).